query/src/sql/planner/binder/limit.rs

```
@@ -0,0 +1,67 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_ast::ast::Expr;
+use common_exception::ErrorCode;
+use common_exception::Result;
+
+use crate::sql::binder::Binder;
+use crate::sql::optimizer::SExpr;
+use crate::sql::planner::semantic::TypeChecker;
+use crate::sql::plans::LimitPlan;
+use crate::sql::BindContext;
+
+impl<'a> Binder {
+    pub(super) async fn bind_limit(
+        &mut self,
+        limit: Option<&Expr<'a>>,
+        offset: &Option<Expr<'a>>,
+        bind_context: &mut BindContext,
+    ) -> Result<()> {
+        let type_checker = TypeChecker::new(bind_context, self.ctx.clone());
+
+        let limit_cnt = match limit {
+            Some(Expr::Literal(x)) => Some(
+                type_checker
+                    .parse_literal(x, None)
+                    .unwrap()
```

Avoid using `unwrap`, let's return an error, or sets a default value.

---

ok